internal/filter/filter.go

```diff
@@ -216,6 +216,27 @@ func match(pattern Pattern, strs []string) (matched bool, err error) {
 	return false, nil
 }
 
+// ValidatePatterns validates a slice of patterns.
+// It returns true if all patterns are valid - false otherwise, along with the invalid patterns.
+func ValidatePatterns(patterns []string) (allValid bool, invalidPatterns []string) {
+	invalidPatterns = make([]string, 0)
+
+	for _, pattern := range patterns {
```

Comment:
Wouldn't it be better to use `ParsePatterns` or `preparePattern` to preparse the pattern first?

Reply:
Changed it accordingly 👍🏻